File: src/main/java/net/reflact/engine/commands/GamemodeCommand.java
package net.reflact.engine.commands;

import net.minestom.server.command.builder.Command;
import net.minestom.server.command.builder.arguments.ArgumentEnum;
import net.minestom.server.command.builder.arguments.ArgumentType;
import net.minestom.server.entity.GameMode;
import net.minestom.server.entity.Player;

public class GamemodeCommand extends Command {
    public GamemodeCommand() {
        super("gamemode", "gm");

        var modeArg = ArgumentType.Enum("mode", GameMode.class).setFormat(ArgumentEnum.Format.LOWER_CASED);

        addSyntax((sender, context) -> {
            if (!(sender instanceof Player player)) return;
            GameMode mode = context.get(modeArg);
            player.setGameMode(mode);
            player.sendMessage("Gamemode set to " + mode);
        }, modeArg);
    }
}

File: src/main/kotlin/net/reflact/engine/commands/GamemodeCommand.kt
package net.reflact.engine.commands

import net.minestom.server.command.builder.Command
import net.minestom.server.command.builder.arguments.ArgumentEnum
import net.minestom.server.command.builder.arguments.ArgumentType
import net.minestom.server.entity.GameMode
import net.minestom.server.entity.Player

class GamemodeCommand : Command("gamemode", "gm") {
    init {
        val modeArg = ArgumentType.Enum("mode", GameMode::class.java).setFormat(ArgumentEnum.Format.LOWER_CASED)

        addSyntax({ sender, context ->
            if (sender !is Player) return@addSyntax
            val mode = context.get(modeArg)
            sender.gameMode = mode
            sender.sendMessage("Gamemode set to $mode")
        }, modeArg)
    }
}
